src/components/MultiHopTrade/components/MultiHopTradeConfirm/hooks/useIsApprovalInitiallyNeeded.tsx

```diff
@@ -1,55 +1,46 @@
 import type { AccountId } from '@shapeshiftoss/caip'
 import { useEffect, useState } from 'react'
-import { checkApprovalNeeded } from 'components/MultiHopTrade/hooks/useAllowanceApproval/helpers'
-import { useWallet } from 'hooks/useWallet/useWallet'
 import type { TradeQuoteStep } from 'lib/swapper/types'
 import { selectFirstHopSellAccountId } from 'state/slices/selectors'
 import {
-  selectActiveQuote,
   selectFirstHop,
   selectSecondHop,
   selectSecondHopSellAccountId,
 } from 'state/slices/tradeQuoteSlice/selectors'
-import { useAppSelector } from 'state/store'
+import { tradeQuoteSlice } from 'state/slices/tradeQuoteSlice/tradeQuoteSlice'
+import { useAppDispatch, useAppSelector } from 'state/store'
+
+import { useIsApprovalNeeded } from './useIsApprovalNeeded'
 
 const useIsApprovalInitiallyNeededForHop = (
   tradeQuoteStep: TradeQuoteStep | undefined,
   sellAssetAccountId: AccountId | undefined,
 ) => {
-  const tradeQuote = useAppSelector(selectActiveQuote)
-  const [isLoading, setIsLoading] = useState<boolean>(true)
-  const [isApprovalInitiallyNeeded, setIsApprovalInitiallyNeeded] = useState<boolean>(false)
-  const wallet = useWallet().state.wallet
-
-  useEffect(() => {
-    ;(async () => {
-      setIsLoading(true)
+  const [watchIsApprovalNeeded, setWatchIsApprovalNeeded] = useState<boolean>(true)
+  const [isApprovalInitiallyNeeded, setIsApprovalInitiallyNeeded] = useState<boolean | undefined>()
 
-      if (!wallet || !sellAssetAccountId || !tradeQuoteStep) {
-        return
-      }
+  const { isLoading, isApprovalNeeded } = useIsApprovalNeeded({
+    tradeQuoteStep,
+    sellAssetAccountId,
+    watch: watchIsApprovalNeeded,
+  })
 
-      if (!tradeQuoteStep) {
-        setIsApprovalInitiallyNeeded(false)
-        setIsLoading(false)
-        return
-      }
-
-      const updatedIsApprovalNeeded = await checkApprovalNeeded(
-        tradeQuoteStep,
-        wallet,
-        sellAssetAccountId,
-      )
-
-      setIsApprovalInitiallyNeeded(updatedIsApprovalNeeded)
-      setIsLoading(false)
-    })()
-  }, [isApprovalInitiallyNeeded, sellAssetAccountId, tradeQuote, tradeQuoteStep, wallet])
+  useEffect(() => {
+    // stop polling on first result
+    if (!isLoading && isApprovalNeeded !== undefined) {
+      setWatchIsApprovalNeeded(false)
```

Comment:
Not sure about this one as well but let's tackle it in a follow-up @woodenfurniture.
Digging into the implementation of this, looks like ths only goal is:
1. to early return from refetching `useAllowance` query (can not refetch and use `enabled` instead of refetching)
2. to use as `enabled` in `useBlockNumber` hook 

I can see how the latter is problematic because of circular deps, since `useAllowance` has a dependency on `blockNumber`, but `blockNumber` itself should stop watching once we have an allowance.

One way to fix this and make things cleaner would be to have a state field in `useAllowance`, set the allowance when we have `data`, and use that as an `enabled` arg for `useBlockNumber`.

Reply:
As discussed will tackle this as follow-up with this issue #3331